codex-rs/tui/src/bottom_pane/mod.rs

```diff
@@ -80,6 +80,30 @@ impl BottomPane<'_> {
         }
     }
 
+    /// Handle Ctrl-C in the bottom pane. If a modal view is active it gets a
+    /// chance to consume the event (e.g. to dismiss itself). Returns true when
+    /// handled.
+    pub(crate) fn on_ctrl_c(&mut self) -> bool {
```

Comment:
I would consider introducing a tiny `enum` type to make the contract self-documenting, like:

```rust
enum CancellationEvent {
    Ignored,
    Handled,  // or maybe Consumed
}
```

and using it as the return value of this function.